modules/minileven/theme/pub/minileven/footer.php

```diff
@@ -30,9 +30,9 @@
 
 <?php
 	global $wp;
-	$current_url =  trailingslashit( home_url( add_query_arg( array(), $wp->request ) ) );
+	$current_url =  home_url( add_query_arg( array( 'ak_action' => 'reject_mobile' ), add_query_arg( $_GET, $wp->request ) ) );
```

Comment:
Using the same approach as above:

```php
	global $wp;
	$url_params = array(
		'ak_action' => 'reject_mobile',
	);
	if ( is_array( $_GET ) && ! empty( $_GET ) ) {
		$url_params[] = $_GET;
	}
	$current_url = home_url(
		add_query_arg(
			$url_params,
			$wp->request
		)
	);
```

Reply:
Does wp->request keep _GET parameters? If not, your approach will not work when the original URL contains parameters. Otherwise, I think your approach is OK but
```php
home_url( add_query_arg( array( 'ak_action' => 'accept_mobile' ), $wp->request ) );
```
would be even better.

Reply:
> Does wp->request keep _GET parameters?

This should take care of that:

```
if ( is_array( $_GET ) && ! empty( $_GET ) ) {
	$url_params[] = $_GET;
}
```

Reply:
Hi, I have not been programming in PHP for years so didn't know it works. Now, I check, and it does work. However, I still don't think it is necessary to make things more complicated if it works with two add_query_arg. We can even use `home_url( add_query_arg('ak_action', 'reject_mobile') );` if we want a short solution. More complicated code means more likely to be buggy IMO.

Reply:
> We can even use home_url( add_query_arg('ak_action', 'reject_mobile') ); if we want a short solution

That won't work I am afraid, as you would then be missing the post slug when viewing single posts or pages. It would only be good on the home page.

Reply:
According to my testing, it works viewing single post or page. Could you show me a counter example?

Reply:
Did you test when using only `home_url( add_query_arg('ak_action', 'reject_mobile') ); `, or the patch in this PR?

Reply:
I tested both of them. They worked the same.

Reply:
It does indeed seem to work well in my tests; let's go with that simpler implementation!

Reply:
I have updated it with the simpler impl.

Reply:
I ran into an issue that seemingly arises from omitting `$wp->request` in the call to `add_query_arg`.

Adding that argument alone isn't enough, since query string parameters aren't retained. I've implemented the approach proposed by @jeherve and submitted PR #12414, which describes the issue I encountered.